code/clue.js

```diff
@@ -1,59 +1,363 @@
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    if (!killerIsPicked) {
+    document.getElementById('killerLoader').style.visibility='visible' 
+    setTimeout(function () {
+      document.getElementById('killerLoader').style.visibility='hidden'
+      shuffleFavouriteWeapon(suspects); //This randomly selects a favourite weapon.
+      mystery.killer = randomSelector(suspects);
+      document.getElementById('killerCard').style.background = mystery.killer.color;
+      document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+      document.getElementById('killerAge').innerHTML = mystery.killer.age;
+      document.getElementById('killerOccupation').innerHTML = mystery.killer.occupation;
+      document.getElementById('killerImage').src = mystery.killer.image;
+      document.getElementById('killerDescription').innerHTML = mystery.killer.description;
+      document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+      killerIsPicked = true;
+      }, 2500);
+    }
+};
+
+document.getElementById('killerCard').addEventListener('click', pickKiller);
+
+// Pick weapon function
+const pickWeapon = () => {
+  if (!weaponIsPicked) {
+    document.getElementById('weaponLoader').style.visibility='visible';
+    setTimeout(function () {
+      document.getElementById('weaponLoader').style.visibility='hidden';
+      mystery.weapon = randomSelector(weapons); 
+      document.getElementById('weaponImage').src = mystery.weapon.image;
+      document.getElementById('weaponName').innerHTML = mystery.weapon.name;
+      document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight}kg`;
+      weaponIsPicked = true;
+    }, 2500)
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+};
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
+
+// Pick room function
+const pickRoom = () => {
+    if (!roomIsPicked) {
+    document.getElementById('roomLoader').style.visibility='visible';
+    setTimeout(function () {
+      document.getElementById('roomLoader').style.visibility='hidden';
+      mystery.room = randomSelector(rooms);
+      document.getElementById('roomName').innerHTML = mystery.room.name;
+      document.getElementById('roomImage').src = mystery.room.image;
+      roomIsPicked = true
+    }, 2500)
+  }
+};
+
+document.getElementById('roomcard').addEventListener('click', pickRoom);
+
+
+// Reveal function
+
+//const revealMystery = () => { - this function does not work when created using const keyword
+  function revealMystery() {
```

Comment:
As you have mentioned you want to use arrow notation for functions but it didn't work for you. I don't see what you need to do to get it to work. Wondering if it can have something to do that you do onclick on a html-tag ` <button>`